packages/react-core/src/components/Modal/ModalContent.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ModalBoxBody from './ModalBoxBody';
+import ModalBoxHeader from './ModalBoxHeader';
+import ModalBoxHCloseButton from './ModalBoxCloseButton';
+import ModalBox from './ModalBox';
+import ModalBoxFooter from './ModalBoxFooter';
+import Backdrop from './Backdrop';
+import Bullseye from '../../layouts/Bullseye/Bullseye';
+
+const propTypes = {
+  /** content rendered inside the Modal. */
+  children: PropTypes.node.isRequired,
+  /** additional classes added to the button */
+  className: PropTypes.string,
+  /** Flag to show the modal */
+  show: PropTypes.bool,
+  /** Content of the Modal Header */
+  header: PropTypes.any,
+  /** Content of the Modal Footer */
+  footer: PropTypes.any,
+  /** A callback for when the close button is clicked */
+  onClose: PropTypes.func,
+  /** Creates a large version of the Modal */
+  isLarge: PropTypes.bool,
+  /* Aria label used for Modal Box Header */
+  label: PropTypes.string.isRequired
+};
+
+const defaultProps = {
+  className: '',
+  show: false,
+  header: null,
+  footer: null,
+  onClose: () => undefined,
+  isLarge: false
+};
+
+const ModalContent = ({
+  children,
+  className,
+  show,
+  header,
+  footer,
+  onClose,
+  isLarge,
+  label,
+  ...props
+}) => {
+  const modalBoxHeader = header ? (
+    <ModalBoxHeader> {header} </ModalBoxHeader>
+  ) : null;
+  const modalBoxFooter = footer ? (
+    <ModalBoxFooter> {footer} </ModalBoxFooter>
+  ) : null;
+  if (show) {
```

Comment:
move this up to the top to avoid other effort if the modal is closed
```jsx
  ...props
}) => {
  if (!isOpen) {
    return null;
  }
  // rest of rendering here
}

Reply:
I'd rather use `<React.Fragment>` here, also this would probably throw a lot of errors because there is no `onClose` action for div. Do we really need it here? I guess we don't need to have it and just use Backdrop, so the final render would looks like:
```JSX
<React.Fragment>
  {isOpen && <Backdrop>
        <Bullseye>
          <ModalBox className={className} isLarge={isLarge} label={label}>
            <ModalBoxHCloseButton onClose={onClose} />
              {modalBoxHeader}
            <ModalBoxBody> {children} </ModalBoxBody>
              {modalBoxFooter}
          </ModalBox>
        </Bullseye>
      </Backdrop>}
</React.Fragment>
```

Reply:
Fragments are only needed if you want to return multiple siblings (without using an array or keys).  Returning a fragment here would be unnecessary. It could be done as easily as just return `isOpen && <Backdrop> ...</Backdrop>`

I don't follow for the onClose with the div? the onClose method is removed from the props object with spread.  it would not make it to the div, or anywhere the rest of props are spread.